src/main/java/selenium/pages/BasePage.java

```diff
@@ -0,0 +1,18 @@
+package selenium.pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import selenium.Driver;
+
+public class BasePage {
+    protected WebDriver driver;
+
+    public BasePage () {
+        this.driver = Driver.getInstance();
```

Comment:
better to pass a driver to the BasePage as a parameter in the constructor:  public BasePage (WebDriver driver)

Reply:
Updated